mods/persistence/controllers/subsystems/persistence.dm

```diff
@@ -1,679 +1,278 @@
-			CRASH("SSPersistence: Couldn't establish DB connection during Limbo Addition!")
-		new_db_connection = TRUE
-	. = one_off.AddToLimbo(things, key, limbo_type, metadata, metadata2, modify)
-	if(.) // Clear it from the queued removals.
-		for(var/list/queued in limbo_removals)
-			if(queued[1] == sanitize_sql(key) && queued[2] == limbo_type)
-				limbo_removals -= list(queued)
-	if(new_db_connection)
-		close_save_db_connection()
-
-/datum/controller/subsystem/persistence/proc/RemoveFromLimbo(var/limbo_key, var/limbo_type)
-	var/new_db_connection = FALSE
-	if(!check_save_db_connection())
-		if(!establish_save_db_connection())
-			CRASH("SSPersistence: Couldn't establish DB connection during Limbo Removal!")
-	. = one_off.RemoveFromLimbo(limbo_key, limbo_type)
-	if(new_db_connection)
-		close_save_db_connection()
-
-/datum/controller/subsystem/persistence/proc/DeserializeOneOff(var/limbo_key, var/limbo_type, var/remove_after = TRUE)
-	var/new_db_connection = FALSE
-	if(!check_save_db_connection())
-		if(!establish_save_db_connection())
-			CRASH("SSPersistence: Couldn't establish DB connection during Limbo Deserialization!")
-		new_db_connection = TRUE
-	. = one_off.DeserializeOneOff(limbo_key, limbo_type, remove_after)
-	if(remove_after)
-		limbo_removals += list(list(sanitize_sql(limbo_key), limbo_type))
-	if(new_db_connection)
-		close_save_db_connection()
-
-// Get an object from its p_id via ref tracking. This will not always work if an object is asynchronously deserialized from others.
-// This is also quite slow - if you're trying to locate many objects at once, it's best to use a single query for multiple objects.
-/datum/controller/subsystem/persistence/proc/get_object_from_p_id(var/target_p_id)
-
-	// Check to see if the object has been deserialized from limbo and not yet added to the normal tables.
-	if(target_p_id in limbo_refs)
-		var/datum/existing = locate(limbo_refs[target_p_id])
-		if(existing && !QDELETED(existing) && existing.persistent_id == target_p_id)
-			return existing
-		limbo_refs -= target_p_id
-
-	// If it was in limbo_refs we shouldn't find it in the normal tables, but we'll check anyway.
-	var/new_db_connection = FALSE
-	if(!check_save_db_connection())
-		if(!establish_save_db_connection())
-			CRASH("SSPersistence: Couldn't establish DB connection during Object Lookup!")
-		new_db_connection = TRUE
-	var/DBQuery/world_query = dbcon_save.NewQuery("SELECT `p_id`, `ref` FROM `[SQLS_TABLE_DATUM]` WHERE `p_id` = \"[target_p_id]\";")
-	SQLS_EXECUTE_AND_REPORT_ERROR(world_query, "OBTAINING OBJECT FROM P_ID FAILED:")
-
-	while(world_query.NextRow())
-		var/list/items = world_query.GetRowData()
-		var/datum/existing = locate(items["ref"])
-		if(existing && !QDELETED(existing) && existing.persistent_id == items["p_id"])
-			. = existing
-		break
-
-	if(new_db_connection)
-		close_save_db_connection()
-
-/datum/controller/subsystem/persistence/proc/print_db_status()
-	return SQLS_Print_DB_STATUS()
+/mob/Stat()
```

Comment:
What is this doing, and why is it here?

Reply:
It's printing the last save loaded to admins/server staff in their status panel. It's pretty well documented in the doc.

Reply:
Sorry, I'm referring to the ``/mob/Stat()`` proc below.

Reply:
That's what I was replying to.